src/history/hs_cursor.c

```diff
@@ -178,9 +178,14 @@ __wt_hs_cursor_position(WT_SESSION_IMPL *session, WT_CURSOR *cursor, uint32_t bt
  *     globally. Otherwise, a prepare conflict will be returned upon reading a prepared update.
  */
 static int
-__hs_find_upd_int(WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key,
-  const char *value_format, uint64_t recno, WT_UPDATE_VALUE *upd_value, bool allow_prepare,
-  WT_ITEM *on_disk_buf, WT_TIME_WINDOW *on_disk_tw)
+__hs_find_upd_int(
+  WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key, const char *value_format,
+  uint64_t recno, WT_UPDATE_VALUE *upd_value, bool allow_prepare, WT_ITEM *on_disk_buf
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
Why is this necessary?

Reply:
on_disk_tw is now only used in an assert so production build will complain about unused variable.

Reply:
Is there a reason that we can't use `WT_UNUSED` to get around this?

Reply:
No, I can give that a try.

Reply:
Cool, I'd much prefer that. It should be cleaner if we can get that working.